apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -362,15 +368,18 @@ export const submitChatContents = createAsyncThunk(
         .logError('Error in aichat completion request', error as Error);
 
       thunkAPI.dispatch(clearChatMessagePending());
-      thunkAPI.dispatch(addChatMessage({...newMessage, status: Status.ERROR}));
+      thunkAPI.dispatch(
+        addChatEvent({...newUserMessage, status: Status.ERROR} as ChatEvent)
```

Comment:
do we need the cast here?

Reply:
I added bc I am getting a TS error without the cast.
![Screenshot 2024-08-07 at 4 13 21 PM](https://github.com/user-attachments/assets/5710d307-2db3-4ceb-b7d3-a63bf29892c4)

Reply:
Ah interesting, does it work if you use a generic in the `addChatEvent` function signature?
```
<T extends ChatEvent>(chatEvent: T, thunkAPI) => { ... }
```

Reply:
Oh yeah - that worked. Thanks!